src/databricks/labs/ucx/azure/resources.py

```diff
@@ -435,27 +444,45 @@ def _role_assignment(
         scope = assignment_properties.get("scope")
         if not scope:
             return None
-        role_name = self._role_name(role_definition_id)
+        role_details = self._role_name(role_definition_id)
+        role_name = role_details.role_name
         if not role_name:
             return None
+        role_type = role_details.role_type
+        role_permissions = role_details.role_permissions
         principal = self._get_principal(principal_id)
         if not principal:
             return None
         if scope == "/":
             scope = resource_id
         return AzureRoleAssignment(
-            resource=AzureResource(resource_id), scope=AzureResource(scope), principal=principal, role_name=role_name
+            resource=AzureResource(resource_id),
+            scope=AzureResource(scope),
+            principal=principal,
+            role_name=role_name,
+            role_type=role_type,
+            role_permissions=role_permissions,
```

Comment:
you can assign directly from the dataclass
```suggestion
            role_name=role_details.role_name,
            role_type=role_details.role_type,
            role_permissions=role_details.role_permissions,
```

Reply:
@nkvuong - Thanks, I have added this.